packages/amplify-category-function/src/provider-utils/awscloudformation/service-walkthroughs/scheduleWalkthrough.ts

```diff
@@ -0,0 +1,352 @@
+        choices: ['Update the schedule', 'Remove the schedule'],
+      };
+
+      const scheduleEventOperationAnswer = await inquirer.prompt([scheduleEventOperationQuestion]);
+
+      switch (scheduleEventOperationAnswer.ScheduleEventOperation) {
+        case 'Update the schedule': {
+          // add service walkthrough to get the cron expression
+          let cloudWatchRule = await cronServiceWalkthrough(context);
+          scheduleParams.cloudwatchEnabled = 'true';
+          scheduleParams.cloudwatchRule = cloudWatchRule;
+          fs.writeFileSync(cfnFilePath, JSON.stringify(cfnContent, null, 4));
+          break;
+        }
+        case 'Remove the schedule': {
+          scheduleParams.cloudwatchEnabled = 'false';
+          scheduleParams.cloudwatchRule = 'NONE';
+          delete cfnContent.Resources.CloudWatchEvent;
+          delete cfnContent.Resources.PermissionForEventsToInvokeLambda;
+          delete cfnContent.Outputs.CloudWatchEventRule;
+
+          fs.writeFileSync(cfnFilePath, JSON.stringify(cfnContent, null, 4));
+          break;
+        }
+        default:
+          console.log(`${scheduleEventOperationAnswer.scheduleEventOperation} not supported`);
+      }
+    }
+  }
+  return scheduleParams;
+}
+
+async function cronServiceWalkthrough(context: any) {
+  let cloudwatchRule;
+  // resource questions for setting cron
+  const intervalQuestion = {
+    type: 'list',
+    name: 'interval',
+    message: 'At which interval should the function be invoked?',
+    choices: ['minutes', 'hourly', 'daily', 'weekly', 'monthly', 'yearly', 'customRule'],
+  };
+  const intervalAnswer = await inquirer.prompt([intervalQuestion]);
+  switch (intervalAnswer.interval) {
+    case 'minutes': {
+      const minuteQuestion = {
+        type: 'input',
+        name: 'minutes',
+        message: 'Enter the rate in mintues?',
+        validate: context.amplify.inputValidation({
+          validation: {
+            operator: 'regex',
+            value: '^[1-9][0-9]*$',
+            onErrorMsg: 'Resouce should be numeric',
+          },
+          required: true,
+        }),
+      };
+      const minuteAnswer = await inquirer.prompt([minuteQuestion]);
+      if (minuteAnswer.minutes === '1') {
+        cloudwatchRule = 'rate(' + minuteAnswer.minutes + ' minute)';
+      } else {
+        cloudwatchRule = 'rate(' + minuteAnswer.minutes + ' minutes)';
+      }
+      break;
+    }
+    case 'hourly': {
+      const hourQuestion = {
+        type: 'input',
+        name: 'hours',
+        message: 'Enter the rate in hours?',
+        validate: context.amplify.inputValidation({
+          validation: {
+            operator: 'regex',
+            value: '^[1-9][0-9]*$',
+            onErrorMsg: 'Resouce should be numeric',
+          },
+          required: true,
+        }),
+      };
+      const hourAnswer = await inquirer.prompt([hourQuestion]);
+      if (hourAnswer.hours === '1') {
```

Comment:
simplifies to
```javascript
const plural = hourAnswer.hours === 1 ? '' : 's';
cloudwatchRule = `rate(${hourAnswer.hours} hour${plural}`
```

Reply:
thanks for the nice piece of code. replaced